micrometer-jmx/src/main/scala/com/avast/sst/micrometer/jmx/TypeScopeNameObjectNameFactory.scala

```diff
@@ -0,0 +1,51 @@
+package com.avast.sst.micrometer.jmx
+
+import java.util
+import java.util.regex.Pattern
+
+import com.codahale.metrics.jmx.{DefaultObjectNameFactory, ObjectNameFactory}
+import javax.management.ObjectName
+
+import scala.util.Try;
+
+/** This is custom [[com.codahale.metrics.jmx.ObjectNameFactory]] which uses "type-scope-name" hierarchy of resulting
+  * [[javax.management.ObjectName]] (levels 3-N are glued together).
+  */
+class TypeScopeNameObjectNameFactory(separator: String = ".") extends ObjectNameFactory {
+
+  private val quotedSeparator = Pattern.quote(separator)
+
+  private val defaultFactory = new DefaultObjectNameFactory()
+
+  private val partNames = Vector("type", "scope", "name")
+
+  override def createName(`type`: String, domain: String, name: String): ObjectName = {
+    val parsedName = parseName(domain, name)
+    parsedName.getOrElse(defaultFactory.createName(`type`, domain, name))
+  }
+
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+  private def parseName(domain: String, name: String): Try[ObjectName] = Try {
```

Comment:
`Try` (as everything Scala related from Twitter, like `Future` or hysterical community drama) come with more headache than worth it
It literally catches everything, which is almost always problematic. Could you perhaps catch only the expected exceptions (or `NonFatal`, or use `Either.catchNonFatal` or whatever it's called)?

Reply:
I don't think it could cause any real trouble here but yeah, I'll replace it with `Either`.